tomcat/src/main/tomcat-base/conf/catalina.properties

```diff
@@ -157,3 +157,6 @@ session.DatastoreStore.sessionKind=TomcatGCloudSession
 
 # Defines the maximum time (in seconds) a session can be inactive before being deleted by the expiration process.
 session.DatastoreStore.sessionMaxInactiveTime=3600
+
+gcp.stackdriver-trace.scheduledDelay=15
+gcp.project=""
```

Comment:
If the project property is not set the TraceValve will throw a LifecycleException and stop Tomcat initialization.

On GCP the project is deduced from the environment but can be overridden by the user through `TOMCAT_PROPERTIES`.

The default behavior for the `cloud-trace-java` library when a empty or invalid project is specified is to fail silently.